### ISSUE TITLE
3.0.1b explain/explain analyze

### ISSUE THREAD

**Issue by pgaro** [OmniDB/OmniDB#1170]
First it's obvious that there are no shortcuts anymore for this, I wonder about the decision. Not that it would matter much considering the variety of issues below.

What surprises me even more is the new view. While it's certainly nice to have a tree view, it doesn't really seem well equipped to do what it's supposed to. When I open this a Single node almost fills my screen, even when I hide the node data I can fit at most two nodes on my screen without rescaling and at the point I'm fitting 5 it gets hard to read the values. The nodes itself are very narrow, the spacing is ok, if there is no subtree moving them apart. Maybe there could be a more intelligent way to prevent the subtrees from colliding than moving them to different columns?

My main complaint though is the fact that it's so incredibly hard to navigate within the tab. The scrollbar at the bottom of the tab is rather hard to grab, so the column thing is an issue. I just looked at a simple explain analyze and clicked the "show me the graph button" and this is my result 
![Explain_Plan](https://user-images.githubusercontent.com/38760999/97309361-3598cc80-1862-11eb-8c85-00466329bf5e.jpg)
That is literally what I saw. Do you think that view is helpful in any way?

it becomes more hilarious when on deals with partitioned tables, let me humor you with a very simple join between two of them in my test database
![ExplainPartitioned](https://user-images.githubusercontent.com/38760999/97312919-55ca8a80-1866-11eb-8b03-859f39e7461c.jpg)
While it looks neat, I think that is probably the reason displaying the nodes in a direction one can't even scroll, makes the view utterly useless.

There is no organic way of resizing (ctrl+mouse wheel is the first thing my coworkers tried) to it's always a lot of fiddling until I get a useful size and then scrolling to interesting part of the graph. What even makes this process worse is that during scrolling the left upper corner of the area is fixed, instead of the center of the screen, which sadly doesn't seem trivial to do.

Then there appears to be a bug with the toggle details button, because at some layer (3?) the nodes buttons are suddenly ignored - toggling all still works.

Furthermore I'd like to complain about the naming of the nodes. We know that both group and hash aggregates are aggregates. But the fact that I need to open the details to read that basic information what kind of aggregate I'm dealing with, makes it almost impossible to read.

I'm also surprised at the way SubPlans and CTEs work. They usually provide a very helpful way to navigate along the plan, because the name tells me where in the query I have to look. Your node view doesn't have any of that.

I could go on for a while, but let me close with just one other complaint: The thing I miss a lot coming from the explain feature of the OmniDB 2 is the option to simple toggle the visibility of whole subtrees. In my view that was the main strength of the old explain plan, while the coloring was a close second.

tl;dr I think as it is, the node view is almost unusable unless you have just 4 nodes, in which case it has no benefit over the plain explain plan output. I see no trivial way of fixing this, so I want to ask whether you would consider making it optional to a view that works similar to the old explain. It was way simpler, but actually usable. I care less about getting all of these things fixed, I think that's just too much, but rather have a worker alternative. Getting a usable tree view comes with a ton of very hard UX problems, so personally wouldn't attempt to do it.

**Comment by rafaelthca** [comment]
Hi @pgaro,

Thanks for raising those concerns. I'm already aware of all of this, since I've
been using myself in my daily work.

The idea of coming up with a new component was due to the fact that we were missing
a more comprehensive graphical view of the plan. The old component lacks several
details in the graph view, which are present in the new one.

That being said, all the issues you reported make the component very hard to use
in complex plans.

We made the decision to bring back the old component and will keep both. People
will be able to use the old one by default and optionally switch to the new one,
which will be incrementally improved.

Users that make extensive use of the graphical explain cannot be penalized by
not having this feature until we are able to fix all the reported problems.

You will see the old component when we release the new hotfix version, 3.0.2b.

Again, thanks for reporting all the problems, as you were doing in 2.x as well.

Rafael.

**Comment by pgaro** [comment]
@rafaelthca Thank you for work, reporting is the least I can do, when I don't want to get my hands dirty in front end code. If you ever want to get feedback about something, you can just get in touch. I have a few people I can ask to try something out, if I need input I'm not able to give myself.

**Comment by pgaro** [comment]
This is much better, even though I still miss the shortcuts for explain/explain analyze.

**Comment by rafaelthca** [comment]
Hi,

The shortcuts were added back. Can you confirm they are working?

**Comment by pgaro** [comment]
Sorry, my bad. I missed that part. That's helpful, thank you!

I understand that you were trying to move every shortcut to some Alt+\w combination, but is there a particular reason you changed the explain shortcuts with the ident and such?

I can relearn or reconfigure them at my leisure, but I'm not sure what other more habitual users will think.

**Comment by felipebiox** [comment]
Hi @pgaro 

As mentioned in another issue about shortcuts, we created a pinned issue (#1196) to centralize the list of desired shortcuts.

I'm closing this issue and adding the shortcuts in the list there.

Best regards
